lib/widgets/weekday.dart

```diff
@@ -0,0 +1,109 @@
+          Text(
+            "${formatter.format(course?.start?.toLocal())} - ${formatter.format(course?.end?.toLocal())} | ",
+          ),
+          Text((course == null || course?.location == '')
+              ? "N/A"
+              : course.location)
+        ]),
+        trailing: Row(mainAxisSize: MainAxisSize.min, children: <Widget>[
+          RaisedButton(
+              onPressed: (course != null && course?.location != '')
+                  ? () {
+                      String letter = course.location[0];
+                      if (letter == "H") {
+                        Provider.of<MapData>(context, listen: false)
+                            .changeCampus('sgw');
+                        Provider.of<MapData>(context, listen: false)
+                            .changeEnd(constants.hBuilding);
+                        Provider.of<MapData>(context, listen: false)
+                                .controllerDestination
+                                .text =
+                            course.location[1] == "A"
+                                ? "Hall Building, Montreal"
+                                : course.location;
+                      } else if (letter == "M") {
+                        Provider.of<MapData>(context, listen: false)
+                            .changeCampus('sgw');
+                        Provider.of<MapData>(context, listen: false)
+                            .controllerDestination
+                            .text = course.location;
+                        Provider.of<MapData>(context, listen: false)
+                            .changeEnd(constants.jmsbBuilding);
+                      } else if (letter == "L") {
+                        Provider.of<MapData>(context, listen: false)
+                            .changeCampus('loyola');
+                        Provider.of<MapData>(context, listen: false)
+                            .controllerDestination
+                            .text = "Loyola Campus, Montreal";
+                        Provider.of<MapData>(context, listen: false)
+                            .changeEnd(constants.loyola);
+                      } else if (letter == "J") {
+                        Provider.of<MapData>(context, listen: false)
+                            .changeCampus('sgw');
+                        Provider.of<MapData>(context, listen: false)
+                            .controllerDestination
+                            .text = "John Molson Business, Montreal";
+                        Provider.of<MapData>(context, listen: false)
+                            .changeEnd(constants.jmsbBuilding);
+                      } else if (letter == "F") {
+                        Provider.of<MapData>(context, listen: false)
+                            .changeCampus('sgw');
+                        Provider.of<MapData>(context, listen: false)
+                            .controllerDestination
+                            .text = "FG Building, Montreal";
```

Comment:
Data like (F - sgw - FG Building, Montreal) is also being used in location_search_dart.
Are we planning on having building location data externalized instead of hardcoded?

Reply:
Agreed, we should probably rely on the global search instead of interpreting destinations in different parts of the code

Reply:
This'll have to be done during refactoring.

Reply:
@andrelegault  Sprint 4 you mean? If so, close the conversation.

Reply:
yeah i meant sprint 4